<?xml version="1.0" ?>
<launch>
  <!--Be sure to pass the IPs of your pandas like robot_ips:={panda_1/robot_ip: <my_ip_1>, panda_2/robot_ip: <my_ip_2>}   -->
  <arg name="robot1_ip" />
  <arg name="robot2_ip" />

  <arg name="robot_id" default="panda_dual" />
  <arg name="rviz" default="true" />
  <arg name="rqt" default="true" />

  <include file="$(find franka_control)/launch/franka_combined_control.launch" >
    <arg name="robot_id" value="$(arg robot_id)" />
    <!--arg name="robot_ips" value="$(arg robot_ips)" /!-->
    <arg name="robot_ips" value="{panda_1/robot_ip: $(arg robot1_ip), panda_2/robot_ip: $(arg robot2_ip)}" />
  </include>

  <include ns="panda_1" file="$(find franka_gripper)/launch/franka_gripper.launch">
    <arg name="robot_ip" value="$(arg robot1_ip)" />
  </include>

  <include ns="panda_2" file="$(find franka_gripper)/launch/franka_gripper.launch">
    <arg name="robot_ip" value="$(arg robot2_ip)" />
  </include>

  <group ns="$(arg robot_id)">
    <rosparam command="load" file="$(find panda_cartesian_impedance)/config/cartesian_impedance_controllers.yaml" />
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"  args="dual_arm_cartesian_impedance_controller"/>
    <node name="dual_pose_commander" pkg="panda_cartesian_impedance" type="dual_pose_commander.py" required="true" output="screen" >
        <param name="left_link_name" value="panda_1_link0" />
        <param name="right_link_name" value="panda_2_link0" />
        <remap from="equilibrium_pose" to="dual_arm_cartesian_impedance_controller/dual_arm_target_pose" />
    </node>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" if="$(arg rqt)"/>
    <!--node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_example_controllers)/launch/rviz/franka_dual_description_with_marker.rviz" if="$(arg rviz)"/!-->
  </group>
</launch>
